pkg/istio-agent/xds-proxy.go

```diff
@@ -214,6 +230,9 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 					return err
 				}
 			}
+		case <-p.resetChan:
+			_ = upstream.CloseSend()
+			return nil
```

Comment:
i just remembered. do we need to do something to make sure the go routines in this func terminate?

Reply:
upstream go routine will go away - downstream one we need to see, I will take a look.